docs/src/main/asciidoc/rest-client.adoc

```diff
@@ -432,6 +432,113 @@ You should see a JSON object containing some basic information about Greece.
+In this section we'll demonstrate how Wiremock can be leveraged for testing the `CountriesService` which was developed above.
+
+First of all, Wiremock needs to be added as a test dependency. For a Maven project that would happen like so:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.github.tomakehurst</groupId>
+    <artifactId>wiremock-jre8</artifactId>
+    <scope>test</scope>
+    <version>${wiremock.version}</version> <1>
+</dependency>
+----
+
+<1> Use a proper Wiremock version. All available versions can be found link:https://mvnrepository.com/artifact/com.github.tomakehurst/wiremock-jre8[here].
+
+In Quarkus tests when some service needs to be started before the Quarkus tests are ran, we utilize the `@io.quarkus.test.common.QuarkusTestResource`
+annotation to specify a `io.quarkus.test.common.QuarkusTestResourceLifecycleManager` which can start the service and supply configuration
+values that Quarkus will use.
+
+[NOTE]
+====
+For more details about `@QuarkusTestResource` refer to  link:getting-started-testing#quarkus-test-resource[this] part of the documentation.
+====
+
+Let's create an implementation of `QuarkusTestResourceLifecycleManager` called `WiremockCountries` like so:
+
+[source,java]
+----
+package org.acme.rest.client;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*; // <1>
+
+public class WiremockCountries implements QuarkusTestResourceLifecycleManager {  // <2>
+
+	private WireMockServer wireMockServer;
+
+	@Override
+	public Map<String, String> start() {
+		wireMockServer = new WireMockServer();
+		wireMockServer.start(); // <3>
+
+		stubFor(get(urlEqualTo("/v2/name/GR"))   // <4>
+				.willReturn(aResponse()
+						.withHeader("Content-Type", "application/json")
+						.withBody( // <5>
+						"""
+						[
+							{
+								"name": "Ελλάδα",
+								"capital": "Αθήνα"
+							}
+						]
+						"""
+						)));
+
+		stubFor(get(urlMatching(".*")).atPriority(10).willReturn(aResponse().proxiedFrom("https://restcountries.eu/rest")));   // <6>
+
+		return Collections.singletonMap("org.acme.getting.started.country.CountriesService/mp-rest/url", wireMockServer.baseUrl()); // <7>
+	}
+
+	@Override
+	public void stop() {
+		if (null != wireMockServer) {
+			wireMockServer.stop();  // <8>
+		}
+	}
+}
+----
+
+<1> Statically importing the methods in the Wiremock package makes it easier to read the test.
+<2> The `start` method is invoked by Quarkus before any test is run and returns a `Map` of configuration properties that apply during the test execution.
+<3> Launch Wiremock.
+<4> Configure Wiremock to stub the calls to `/v2/name/GR` by returning a specific canned response.
+<5> This examples uses Text Blocks from Java 14 because it makes the code a lot easier to read.
```

Comment:
I think that is a bad idea.

Reply:
You prefer having crappy json in the document?

Reply:
Considering it's very short and simple, I would yes. At least as long as we target older versions of Java.

Reply:
OK, fixed